Jobs.Fetcher.Facebook/Client/Fetcher.cs

```diff
@@ -278,34 +291,66 @@ public class Fetcher {
             /**
                Fetch children on the `edge` api api path of the social media artifact `node`.
              */
+            if (!node.TryGetValue("id", out var id)) {
+                Logger.Warning($"Could get ID from node of {edge.Name}");
+                return;
+            }
             JObject tip;
             try {
-                tip = FetchEndpoint(edge, node["id"].ToString(), edge.Name, edge.Columns.Select(x => x.Value).ToList(), edge.Ordering).Result;
-            } catch (FacebookApiException) {
+                tip = FetchEndpoint(edge, id.ToString(), edge.Name, edge.Columns.Select(x => x.Value).ToList(), edge.Ordering).Result;
+            } catch (FacebookApiException e) {
+                Logger.Warning($"Couldn't fetch data from edge {edge.Name} due to API exception.");
+                Logger.Debug($"Error: ({e}).");
+                return;
+            } catch (Exception e) {
+                Logger.Warning($"Couldn't fetch data from edge {edge.Name}.");
+                Logger.Debug($"Error: ({e}).");
                 return;
             }
             var result = PaginateEndpoint(edge, tip, max_iters);
             var result2 = new List<JObject>();
-            DatabaseManager.Transactional(connection => {
-                foreach (JObject obj in result) {
-                    var o = obj;
-                    obj.Add(edge.Source.Name + "_id", node["id"]);
-                    AddSystime(edge, ref o);
-                    if (!DatabaseManager.CheckEdgeMatch(connection, edge, o)) {
-                        DatabaseManager.InsertRow(connection, edge, o);
-                    } else {
-                        if (edge.Ordering == "desc") {
-                            break;
-                        }
-                    }
-                    result2.Add(o);
-                }
-                if (result.Count() > 0) {
-                    if (edge.Ordering == "unordered") {
-                        DatabaseManager.DeleteEdgeNotMatch(connection, edge, result2, "systime", "fetch_time");
+
+            var transaction_batch = new List<JObject>();
+            var remaining = result.Count();
+            Logger.Debug($"Adding {remaining} rows to Database.");
+            var batch_number = 0;
+            foreach (JObject batch_obj in result) {
+                transaction_batch.Add(batch_obj);
+                if (transaction_batch.Count >= BATCH_SIZE || transaction_batch.Count >= remaining) {
+                    Logger.Debug($"Adding batch {batch_number} to Database.");
```

Comment:
Just a comment, I met this problem when coding myself, and think we should standardize this. Do you think we should count starting from zero or from 1, when printing those messages? I like to start with 1, and then use the format "batch X of Y", I feel like it gives a bit more context

Reply:
I like it. So let's always start with 1 and let's always present the total number of batches